tests/journald_tests.cpp

```diff
@@ -292,6 +293,189 @@ TEST_F(JournaldLoggerTest, ROOT_LogToJournaldWithBigLabel)
 }
 
 
+// This checks a specific case where the arguments passed into the
+// ContainerLogger will differ if the Mesos agent restarts before
+// launching a nested container.
+TEST_F(JournaldLoggerTest, ROOT_CGROUPS_LaunchThenRecoverThenLaunchNested)
+{
+  mesos::internal::slave::Flags flags = CreateSlaveFlags();
+  flags.launcher = "linux";
+  flags.isolation = "cgroups/cpu,filesystem/linux,namespaces/pid";
+
+  // Use the journald container logger.
+  flags.container_logger = JOURNALD_LOGGER_NAME;
```

Comment:
All journald logger tests (unless modified otherwise) use the combo logrotate+journald setting.  We run into the problem regardless because the logrotate vs journald setting only has an effect in the companion binary, rather than the logger module loaded into the agent.